trackpy/linking.py

```diff
@@ -515,13 +555,20 @@ def link_df(features, search_range, memory=0,
     if retain_index:
         orig_index = features.index.copy()  # Save it; restore it at the end.
     features.reset_index(inplace=True, drop=True)
-    levels = (_build_level(frame, pos_columns, t_column) for frame_no, frame
+    levels = (_build_level(frame, pos_columns, t_column,
+                           diagnostics=diagnostics) for frame_no, frame
               in features.groupby(t_column))
     labeled_levels = link_iter(
         levels, search_range, memory=memory, predictor=predictor,
+        adaptive_step=adaptive_step, adaptive_limit=adaptive_limit,
         neighbor_strategy=neighbor_strategy, link_strategy=link_strategy,
         hash_size=hash_size, box_size=box_size)
 
+    if diagnostics:
+        features = strip_diagnostics(features)
```

Comment:
Could you add a comment noting that `strip_diagnostics` does not modify the original? I was worried for a moment that it did.